### ISSUE TITLE
Upgrade react-native

### ISSUE THREAD

**Issue by steffenkleinle** [digitalfabrik/lunes-app#1010]
**Task**
Upgrade react-native to the latest stable version.

TODOs:

- [x] Upgrade react-native to the latest stable version. If there are to many changes, feel free to split it up into multiple upgrades (e.g. to 0.71 and then to 0.73).
- [ ] Upgrade react-native-header-buttons requiring react-native v0.71: https://github.com/vonovak/react-navigation-header-buttons/releases/tag/v11.0.0
- [x] Upgrade react-native-reanimated, react-native-gesture-handler and react-native-screens to latest which are blocked by this
- [x] Uncomment the findAllByRole section in ImageCarousel.spec.tsx and make it work again (it seems like react-native is migrating from `accessibilityRole` to `role` which let the test fail)
- [x] Remove the patch for react-native-camera that was added in https://github.com/digitalfabrik/lunes-app/pull/404 because of deprecated prop-types (if #873 isn't done before)
- [x] remove workaround in podfile (https://github.com/facebook/react-native/issues/37748)

**Additional context**
Possibly blocked by #1012/https://github.com/digitalfabrik/lunes-app/issues/873.
The upgrade helper might help: https://react-native-community.github.io/upgrade-helper/?from=0.70.15&to=0.73.6

Related: #983.

**Comment by steffenkleinle** [comment]
Done in #873  and #1035
